conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java

```diff
@@ -139,24 +145,68 @@ public void sendCustomPlaybackEvent(String name, Map<String, Object> attributes)
+     * no longer ensure that the session is managed at the correct time. Additional: Since some metadata attributes
+     * relies on the players source we can't ensure that all metadata attributes are present at session creation.
+     * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
+     * <p>
+     * If no source was loaded this method will throw an error.
+     */
+    public void initializeSession() throws ConvivaAnalyticsException {
+        initializeSession(null);
     }
 
-    private void customEvent(BitmovinPlayerEvent event, Map<String, Object> attributes) {
-        if (!isValidSession()) {
+    /**
+     * Initializes a new conviva tracking session.
+     * <p>
+     * Warning: The integration can only be validated without external session managing. So when using this method we can
+     * no longer ensure that the session is managed at the correct time. Additional: Since some metadata attributes
+     * relies on the players source we can't ensure that all metadata attributes are present at session creation.
+     * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
+     *
+     * @param assetName Will be used as contentMetadata.assetName if no source was loaded before. This overrides the
+     *                  source assetName. If no source was loaded and no assetName is present this method will throw an
+     *                  error.
+     */
+    public void initializeSession(String assetName) throws ConvivaAnalyticsException {
+        if (isSessionActive()) {
             return;
         }
 
-        String eventName = event.getClass().getSimpleName();
-        sendCustomPlaybackEvent("on" + eventName, attributes);
+        if (bitmovinPlayer.getConfig().getSourceItem() == null && assetName == null) {
+            throw new ConvivaAnalyticsException("AssetName is missing. Provide assetName attribute or load player source first");
+        }
+
+        if (assetName != null) {
+            metadataOverrides.assetName = assetName;
+        }
+
+        internalInitializeSession();
+    }
+
+    /**
+     * Ends the current conviva tracking session.
+     * Results in a no-opt if there is no active session.
+     * <p>
+     * Warning: The integration can only be validated without external session managing.
+     * So when using this method we can no longer ensure that the session is managed at the correct time.
+     */
+    public void endSession() {
+        if (!isSessionActive()) {
+            return;
+        }
+
+        internalEndSession();
     }
 
     /**
      * Sends a custom deficiency event during playback to Conviva's Player Insight. If no session is active it will NOT
      * create one.
      *
-     * @param message Message which will be send to conviva
+     * @param message  Message which will be send to conviva
```

Comment:
Is this the wanted code style? Having the description of the parameters on the same indent?
In general I think it can introduce a lot of noise in the git history, when extending/renaming parameters.

Reply:
True but it's just the format when auto formatting the code. I can adapt if we have a code style for this?

Reply:
Usually in the code style settings there is a point `Align parameter descriptions`. Just uncheck it. :)